### DISCUSSION TITLE
setFocusHandler to customize onWindowFocus refetch behaviour

### DISCUSSION THREAD

**Discussion by tonystaark** [TanStack/query#1280] [Q&A]
I am currently using useQuery and need to refetchOnWindow focus to true.
However, if I refocus and then I want to click a certain button let say to log out, it will cancel behaviour and do refetch. I have to click the second time on the same button to log out.
How do I configure it so it does not refetch if I click on the button?
I have thought of detecting the element tag to be "BUTTON" , however I am not familiarize in how to use setFocusHandler to configure:

    window.onclick = (e) => {
        if (e.target.tagName !== 'BUTTON') {
             cancelRefetchOnFocusBehaviour()
        }else {
             allows
        }
     }

How do I integrate the above with the setFocusHandler?

**Comment by Andrey-Bazhanov** [comment]
You can try v3 `focusManager` https://react-query-beta.tanstack.com/guides/window-focus-refetching#custom-window-focus-event

**Reply by tonystaark** [reply]
Thank you. But any idea what is the difference between the link u shared and the setFocusHandler I was talking about earlier in my post from
https://react-query.tanstack.com/docs/guides/window-focus-refetching?

It seems they both use focusManager and setFocusHandler but using the same function ? I am confused

**Reply by Andrey-Bazhanov** [reply]
Yep, I just thought that focusManager was introduced in v3

**Reply by Andrey-Bazhanov** [reply]
I think you need call `handleFocus` when you need in your specific case

**Reply by tonystaark** [reply]
Yup, been trying but no luck.. Haha thanks anyway
